manuscript/chapter3.txt

```diff
@@ -7,29 +7,29 @@ C> By [John Haynes](https://www.youtube.com/channel/UCJVXolxwB4x3EsBAzSACCTg) |
+In this metaphor, the password is the key and its "hash value" is the internal dimensions of the lock. To obtain a hash, a complex mathematical algorithm is run against a string of data and the output is an extremely unique character string. Some weaker hash algorithms can produce the same hash for two different sets of data. This is not typically an issue with hashed passwords, and it is safe to say that if you have a hash of a password you have the password in an encrypted state. The password hash is how the password is stored on any modern operating system like Windows, macOS, or Linux or for encrypted containers like BitLocker or encrypted 7-Zip files. With proper security practices, the hash is the only part of the password that will be available for an examiner to inspect, just as a lockpicker might inspect a lock. If you have access to the encrypted item itself, whether it is a physical device, a forensic image (e.g. a .E01 or .dd), or an individual encrypted file file, it is reasonable to attempt to extract the hash value.
 
-Therefore, if the password hash can be extracted, it can be attacked to attempt to crack the password. Hashing algorithms are mathematically a one-way operation. If someone has a password hash, there is no normal mathematical operation that can be performed to reverse engineer the original plaintext password. Additionally, some hashing algorithms are more difficult to crack than others because the speed of decryption is sacrificed for security. However, the user can guess the potential password, hash it, and then compare the resulting hash against the known hash. If it is a match, then the password is cracked. This would be a very slow method to do manually, but there is software like Hashcat that can be used to automate this process to perform thousands of attempts per second. To make the guessing more difficult, the system can implement what is known as "salt" into the hash to obfuscate the hash and make it more difficult to crack.
+If the password hash can be extracted, it can be used in attempts to crack the password. Unfortunately, hashing algorithms are mathematically a one-way operation. If someone has a password hash, there is no normal mathematical operation that can be performed to reverse engineer the original plaintext password.  Instead, the user can guess the potential password, run the algorithm to hash it, and then compare the resulting hash against the known hash. If the two values match, then the password has been found. This would be an impractically slow manual process, but programs like Hashcat can be used to automate the process, performing thousands of attempts per second.
```

Comment:
Minor formatting: Remove the extra space between sentences.

"...reverse engineer the original plaintext password. Instead, the user can guess the potential password..."